importer/helper.go

```diff
@@ -217,38 +236,41 @@ func loadGenesis(db idb.IndexerDb, in io.Reader) (err error) {
 	return db.LoadGenesis(genesis)
 }
 
-// UpdateAccounting triggers an accounting update.
-func UpdateAccounting(db idb.IndexerDb, frozenCache map[uint64]bool, round uint64, genesisJSONPath string, l *log.Logger) (rounds, txnCount int) {
-	return updateAccounting(db, frozenCache, round, genesisJSONPath, 0, l)
-}
-
-func updateAccounting(db idb.IndexerDb, frozenCache map[uint64]bool, round uint64, genesisJSONPath string, numRoundsLimit int, l *log.Logger) (rounds, txnCount int) {
-	rounds = 0
-	txnCount = 0
+// InitialImport imports the genesis block if needed. Returns true if the initial import occurred.
+func InitialImport(db idb.IndexerDb, genesisJSONPath string, l *log.Logger) bool {
 	state, err := db.GetImportState()
-	maybeFail(err, l, "getting import state, %v", err)
-	if state == nil {
-		state = &idb.ImportState{AccountRound: -1}
+
+	// Only import when the state is not found.
+	if err == idb.ErrorNotInitialized {
+		state.AccountRound = -1
```

Comment:
In general it would be nice to store "next round" instead of "last round" everywhere to avoid these issues.

Reply:
Agreed. Related to the comment above about `yieldTxnQuery`. I've been throwing comments about what I'd like to refactor into this PR and added an entry for this refactoring: https://github.com/algorand/indexer/issues/279